plugins/scanners/scancode/src/main/kotlin/ScanCodeResultParser.kt

```diff
@@ -167,15 +167,13 @@ private fun getSpdxLicenseId(license: JsonNode): String {
     val idFromSpdxKey = license["spdx_license_key"].textValueOrEmpty().toSpdxId(allowPlusSuffix = true)
 
     // For regular SPDX IDs, return early here.
-    if (idFromSpdxKey.isNotEmpty() && !idFromSpdxKey.startsWith(LICENSE_REF_PREFIX)) return idFromSpdxKey
+    if (idFromSpdxKey.isNotEmpty()) return idFromSpdxKey
```

Comment:
Is this change non-functional? E.g. if ScanCode has `LicenseRef-` IDs which do not start with `LicenseRef-scancode-` it wouldn't be, right?

Reply:
> Is this change non-functional?

This is indeed a change that does not (intend to) change behavior.

> E.g. if ScanCode has `LicenseRef-` IDs which do not start with `LicenseRef-scancode-` it wouldn't be, right?

I still believe the change does not change behavior.

Just look at these lines in isolation:

    if (idFromSpdxKey.isNotEmpty() && !idFromSpdxKey.startsWith(LICENSE_REF_PREFIX)) return idFromSpdxKey
    return idFromSpdxKey.takeUnless { it.isEmpty() } // ...

The last line could be rewritten as

    if (idFromSpdxKey.isNotEmpty()) return idFromSpdxKey

So inserted into the first snippet we'd get

    if (idFromSpdxKey.isNotEmpty() && !idFromSpdxKey.startsWith(LICENSE_REF_PREFIX)) return idFromSpdxKey
    if (idFromSpdxKey.isNotEmpty()) return idFromSpdxKey

So the last line here basically is a just relaxed version of the first line, which contains the additional `!idFromSpdxKey.startsWith(LICENSE_REF_PREFIX)` condition. Which means we can drop the first line altogether, as the last line returns anyway if only `idFromSpdxKey` is not empty.

Reply:
I'm not sure if I get it. 

What if `idFromSpdxKey` gets assigned to `LicenseRef-HERE-license` in line `idFromSpdxKey` ?
I expect that previously, the return of the function would start with `LicenseRef-scancode-` while now it doesn't anymore?

Reply:
If `idFromSpdxKey` was "LicenseRef-HERE-license", then in the original code line 170 would *not* have returned as the license starts with "LicenseRef-". But line 172 would have returned "LicenseRef-HERE-license" as it's non-empty.

The new code simplifies this by immediately returning "LicenseRef-HERE-license" as it's non-empty.

Before and after this change, the "scancode" infix is only added if no "LicenseRef-" prefix was present at all.